crates/rspack_plugin_javascript/src/dependency/esm/esm_export_imported_specifier_dependency.rs

```diff
@@ -267,6 +280,12 @@ impl ESMExportImportedSpecifierDependency {
       export_mode.items = Some(items);
       export_mode
     } else {
+      if self.reexport_star_from_external_module(module_graph) {
```

Comment:
So this means that `ModernModuleReexportStarExternalDependency` is only added but not removing the original `ESMExportImportedSpecifierDependency`? what if we just replacing these `ESMExportImportedSpecifierDependency` to `ModernModuleReexportStarExternalDependency`, then we don't need to introduce a new `ExportModeType`

Reply:
I tried to determine whether to use `ModernModuleReexportStarExternalDependency` or `ESMExportImportedSpecifierDependency` here https://github.com/web-infra-dev/rspack/blob/a901d54c78b801bb6ae4854da67bf65a3d03c0f3/crates/rspack_plugin_javascript/src/parser_plugin/esm_export_dependency_parser_plugin.rs#L81 . However, this decision depends on whether the module is externalized, and we need access to the module graph, which isn't possible here.

Therefore, IMO we can't create the correct dependency during parsing. If we attempt to replace a dependency like `finish_modules`, I previously implemented it that way, but it's too late since some logic is already in `finish_modules` and has already been executed. 🤔

Reply:
Which logic already executed, what if we replace it in `finish_modules stage -100`